qa/rpc-tests/wallet_shieldcoinbase.py

```diff
@@ -185,3 +192,19 @@ def verify_locking(first, second, limit):
         sync_mempools(self.nodes[:2])
         self.nodes[1].generate(1)
         self.sync_all()
+
+        print("activate NU5")
+        current_height = self.nodes[0].getinfo()['blocks']
+        self.nodes[0].generate(1300-current_height)
+
+        print("create a unified address")
+        account = self.nodes[0].z_getnewaccount()['account']
```

Comment:
This should not be inlined here. Instead, the test above should be run entirely with a UA, and anywhere that we need UA-specific calls (e.g. calling `z_getnewaccount()` instead of `z_getnewaddress(self.addr_type)`, or replacing `z_getbalance()` with `z_getbalanceforaccount()`) we should do that with an extra class method that is overridden as appropriate by the subclasses in `wallet_shieldcoinbase_sprout.py`, `wallet_shieldcoinbase_sapling.py`, and `wallet_shieldcoinbase_unified.py`. I suspect this will mean that the test cannot currently pass for UAs, until `z_getbalanceforaccount` has been implemented; once I have a PR for that and it has been merged, we can rebase this PR to enable the test to pass.